webpack.config.js

```diff
@@ -107,7 +107,32 @@ module.exports = (env, argv) => {
                 }),
                 // This plugin creates a CycloneDX Software Bill of Materials containing an aggregate of all bundled dependencies.
                 // It needs to be deactivated in watch mode
-                !argv.watch && new CycloneDxWebpackPlugin(cycloneDxWebpackPluginOptions)
+                !argv.watch && new CycloneDxWebpackPlugin(cycloneDxWebpackPluginOptions),
+                // This plugin will raise an error if a client file tries to import a server file
+                {
+                    apply(compiler) {
+                        const serverFiles = path.resolve(__dirname, 'src/server')
```

Comment:
I am not sure this will cover all cases as we have server modules that can be imported from `javascript-modules-library` that needs to be detected as well.

Reply:
Ok! This goes further than this webpack plugin and needs to be addressed at the root: having different export conditions for server vs browser usage

Documentation: https://nodejs.org/api/packages.html#conditional-exports
Example in react library: https://github.com/facebook/react/blob/886c5ad936428f168e50e077bd37fe9472ff8d3e/packages/react/package.json#L25-L28